### PR TITLE
distsql: refactor range splitting logic & comments on int64 boundary

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve?

Refactor range splitting logic & comments on int64 boundary

### What is changed and how it works?

What's Changed: function signature and function comments

How it Works: refactor for better understanding of the function logic

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`: None
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

None

### Release note 

- No release note

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -1093,16 +1093,16 @@ func (e *AnalyzeFastExec) updateCollectorSamples(sValue []byte, sKey kv.Key, sam
 			}
 			idxVals = append(idxVals, v)
 		}
-		var bytes []byte
-		bytes, err = codec.EncodeKey(e.ctx.GetSessionVars().StmtCtx, bytes, idxVals...)
+		var keyBytes []byte
+		keyBytes, err = codec.EncodeKey(e.ctx.GetSessionVars().StmtCtx, keyBytes, idxVals...)
```
Comment (dragonly) [inline executor/analyze.go:1097]: The original name `bytes` conflicts with the imported package `bytes`

**Review by winoros — APPROVED** [review]

File: executor/analyze.go (hunk 1 above)
Comment (winoros) [reply executor/analyze.go:1097]: (there are many places naming a variable as `byte`.:joy:

Comment (dragonly) [reply executor/analyze.go:1097]: Well, that's really a bad news :sob:

**Review by disksing — APPROVED** [review]
fix bot issue

**Pull request merged**